lib/ui/painting/display_list_deferred_image_gpu.cc

```diff
@@ -4,20 +4,49 @@
 
 #include "flutter/lib/ui/painting/display_list_deferred_image_gpu.h"
 
+#include <atomic>
+
+#include "display_list_deferred_image_gpu.h"
+#include "third_party/skia/include/core/SkColorSpace.h"
+
 namespace flutter {
 
-sk_sp<DlDeferredImageGPU> DlDeferredImageGPU::Make(SkISize size) {
-  return sk_sp<DlDeferredImageGPU>(new DlDeferredImageGPU(size));
+sk_sp<DlDeferredImageGPU> DlDeferredImageGPU::Make(
+    const SkImageInfo& image_info,
+    sk_sp<DisplayList> display_list,
+    fml::WeakPtr<SnapshotDelegate> snapshot_delegate,
+    fml::RefPtr<fml::TaskRunner> raster_task_runner,
+    fml::RefPtr<SkiaUnrefQueue> unref_queue) {
+  return sk_sp<DlDeferredImageGPU>(new DlDeferredImageGPU(
+      ImageWrapper::Make(image_info, std::move(display_list),
+                         std::move(snapshot_delegate), raster_task_runner,
+                         std::move(unref_queue)),
+      raster_task_runner));
 }
 
-DlDeferredImageGPU::DlDeferredImageGPU(SkISize size) : size_(size) {}
+DlDeferredImageGPU::DlDeferredImageGPU(
+    std::shared_ptr<ImageWrapper> image_wrapper,
+    fml::RefPtr<fml::TaskRunner> raster_task_runner)
+    : image_wrapper_(std::move(image_wrapper)),
+      raster_task_runner_(std::move(raster_task_runner)) {}
 
 // |DlImage|
-DlDeferredImageGPU::~DlDeferredImageGPU() = default;
+DlDeferredImageGPU::~DlDeferredImageGPU() {
+  auto image_wrapper = std::atomic_load(&image_wrapper_);
```

Comment:
The use of `atomic_load` with `image_wrapper_` is no longer necessary.

`image_wrapper_` is now `const` and is not mutated until the `DlDeferredImageGPU` is destructed.

Reply:
Ahh yeah, done